### DISCUSSION TITLE
How Can I Compute The Closest Distance Between Two Concave Meshes?

### DISCUSSION THREAD

**Discussion by harel55** [bulletphysics/bullet3#3396] [General]
I have two meshes, let's call them A and B, and I would like to use bullet to measure the minimal distance between any pair of of their vertices. Mesh A could theoretically be approximated by a few convex hulls, but mesh B cannot (it represents some fairly rough terrain).

So far, I've been looking at [this pybullet example](https://github.com/bulletphysics/bullet3/blob/master/examples/pybullet/examples/getClosestPoints.py) (though I'll be working in C++ for my actual implementation), and it seems very promising. My concern is that while the documentation assures me that I can load in static concave meshes, and `p.getClosestPoints()` happily accepts those concave meshes as inputs, I have not been thoroughly convinced that bullet isn't taking convex hulls in order to perform the distance calculation. More importantly, I've been looking through the C++ documentation, and I can't find any implementation of the `getClosestPoints()` interface that accepts concave shapes or tri-meshes. I'd appreciate any pointers or assurances about what's happening behind the scenes. Thanks!

**Comment by Chengzhi-Daedalus** [comment]
Any updates?
